Swiftfin/Views/ItemEditorView/Components/RefreshMetadataButton.swift

```diff
@@ -0,0 +1,175 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import JellyfinAPI
+import SwiftUI
+
+extension ItemEditorView {
+
+    struct RefreshMetadataButton: View {
+
+        @StateObject
+        private var viewModel: RefreshMetadataViewModel
+
+        @State
+        private var isPresentingEventAlert = false
+        @State
+        private var error: JellyfinAPIError?
+        @State
+        private var isFinishing = false
+        @State
+        private var isLoading = false
+
+        // MARK: - Initializer
+
+        init(item: BaseItemDto) {
+            _viewModel = StateObject(wrappedValue: RefreshMetadataViewModel(item: item))
+        }
+
+        // MARK: - Body
+
+        var body: some View {
+            Menu {
+                Button {
```

Comment:
I've made this change, but I'm not sold on these icons. These were kind of my 'eh, it works' pick so let me know if you'd prefer something else!

<img width="320" alt="Screenshot 2024-11-11 at 16 54 07" src="https://github.com/user-attachments/assets/80e91e33-0c37-45e0-b0c2-924e3b50a15e">

Reply:
I don't really have any other icons in mind, so these seem okay.

However, it is easy to be confused per what each option does, especially between `Refresh` and `Replace All`. It would be handy if we had a small description in the footer of this button which then has a `Learn More...`, similar to many rows in iOS Settings. I would then want us to pop up a modal that describes each option. I had thought about that as well for the `Playback Quality/Device Profile/Compatibility` options.

Reply:
Do you have a design in mind for that? That feels like something we should create a learnMoreView to keep it consistent across uses. 

I can do that either on this PR or I can add this separately to keep the learnMoreView _(Or a better named view)_ separate and apply it to these two spots (Refresh metadata / customDeviceProfile types)

Reply:
Very rough but this commit in on this PR now if you want to see what I have as an early version. Lemme know if you want me to move this to another PR instead!

![Simulator Screenshot - iPhone 16 Pro - 2024-11-12 at 16 35 08](https://github.com/user-attachments/assets/3d003c56-e24b-4e76-9268-d95eca08748b)

Reply:
Sorry, and here is what the footer looks like:

<img width="324" alt="Screenshot 2024-11-12 at 16 36 39" src="https://github.com/user-attachments/assets/59eb2448-5f80-4ebf-b80c-682fdde73614">

Right now this is basically just a button. I'm sure there's some kind of view extension / section extension that might be better for future usage?

Reply:
I'm happy with the text and the modal but I don't have a strong idea of what the Item | Description should look like. This is my working version but 🤷‍♂️

![Simulator Screenshot - iPhone 16 Pro - 2024-11-13 at 12 12 01](https://github.com/user-attachments/assets/3591dbd4-f85a-4c16-921a-7752621847ae)